src/Stratis.Bitcoin.IntegrationTests/SharedSteps.cs

```diff
@@ -50,14 +50,31 @@ public void MineBlocks(int blockCount, CoreNode node, string accountName, string
 
             var balanceIncrease = balanceAfterMining - balanceBeforeMining;
 
-            this.WaitForNodesToSync(node);
+            this.WaitForNodeToSync(node);
 
             var rewardCoinCount = blockCount * Money.COIN * 50;
-            
+
             balanceIncrease.Should().Be(rewardCoinCount + expectedFees);
         }
 
-        public void WaitForNodesToSync(params CoreNode[] nodes)
+        public void MinePremineBlocks(CoreNode node, string walletName, string walletAccount, string walletPassword)
+        {
+            this.WaitForNodeToSync(node);
+
+            var unusedAddress = node.FullNode.WalletManager().GetUnusedAddress(new WalletAccountReference(walletName, walletAccount));
+            var wallet = node.FullNode.WalletManager().GetWalletByName(walletName);
+            var extendedPrivateKey = wallet.GetExtendedPrivateKeyForAddress(walletPassword, unusedAddress).PrivateKey;
+
+            node.SetDummyMinerSecret(new BitcoinSecret(extendedPrivateKey, node.FullNode.Network));
+            node.GenerateStratisWithMiner(2);
+
+            this.WaitForNodeToSync(node);
+
+            var spendable = node.FullNode.WalletManager().GetSpendableTransactionsInWallet(walletName);
+            spendable.Sum(s => s.Transaction.Amount).Should().Be(Money.COIN * 98000004);
```

Comment:
Also here you could reuse the consensus reward properties (ideally)

Reply:
Fixed!